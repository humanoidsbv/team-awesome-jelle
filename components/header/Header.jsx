import React from 'react';
import PropTypes from 'prop-types';

import MainNavigation from './main-navigation/MainNavigation';

import './header.scss';

const changeBodyClass = isMenuVisible => (
  isMenuVisible ? document.body.classList.remove('test') : document.body.classList.add('test')
);

const Header = ({ isMenuVisible, onToggleMenu }) => (
  <header className="header">
    <div className="header__mobile-controls">
      <a
        className="header__company-logo"
        href="#link"
      >
      team awesome
      </a>

      <button
        className={`
          header__menu-button
          header__menu-button--${isMenuVisible ? 'clicked' : ''}
          `}
        type="submit"
        onClick={() => {
          changeBodyClass(isMenuVisible);
          onToggleMenu();
        }}
      >
        <img
          alt="menu-button"
          className="header__menu-button-icon--open"
          width="20px"
          height="20px"
          src="/static/icons/hamburger.svg"
        />
        <img
          alt="menu-button-close"
          className="header__menu-button-icon--close"
          height="20px"
          width="20px"
          src="/static/icons/close.svg"
        />
      </button>
    </div>
    <MainNavigation
      isMenuVisible={isMenuVisible}
    />
  </header>
);

Header.propTypes = {
  isMenuVisible: PropTypes.bool.isRequired,
  onToggleMenu: PropTypes.func.isRequired
};

export default Header;
