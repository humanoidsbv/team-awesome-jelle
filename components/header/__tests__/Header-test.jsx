import React from 'react';
import { shallow } from 'enzyme';

import Header from '../Header';

import {
  toggleMenu
} from '../../../ducks/header';

const isMenuVisibleSelector = false;

it('Renders Entry Component correctly', () => {
  const wrapper = shallow(
    <Header
      isMenuVisible={isMenuVisibleSelector}
      onToggleMenu={toggleMenu}
    />
  );
  expect(wrapper).toMatchSnapshot();
});
