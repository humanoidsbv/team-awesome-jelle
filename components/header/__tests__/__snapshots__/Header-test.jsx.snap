// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Entry Component correctly 1`] = `
<header
  className="header"
>
  <div
    className="header__mobile"
  >
    <a
      className="company-name"
      href="#link"
    >
      team awesome
    </a>
    <button
      className="
          menu-button
          menu-button--
          "
      onClick={[Function]}
      type="submit"
    >
      <img
        alt="menu-button"
        className="menu-button__icon--open"
        height="20px"
        src="/static/icons/hamburger.svg"
        width="20px"
      />
      <img
        alt="menu-button-close"
        className="menu-button__icon--close"
        height="20px"
        src="/static/icons/close.svg"
        width="20px"
      />
    </button>
  </div>
  <nav
    className="
      main-navigation
      main-navigation--close
    "
  >
    <a
      className="main-navigation__item main-navigation__item--active"
      href="#link"
    >
      Timesheets
    </a>
    <a
      className="main-navigation__item"
      href="#link"
    >
      Team members
    </a>
    <a
      className="main-navigation__item"
      href="#link"
    >
      Projects
    </a>
    <a
      className="main-navigation__item"
      href="#link"
    >
      Clients
    </a>
    <a
      className="main-navigation__item"
      href="#link"
    >
      Documents
    </a>
  </nav>
  <button
    className="main-navigation__profile-button"
    type="button"
  >
    <img
      alt=""
      className="main-navigation__company-logo"
      src="/../../static/icons/1_humanoids_logo.png"
    />
    <img
      alt=""
      className="main-navigation__user-icon"
      height="32px"
      src="/../../static/icons/jelle.jpg"
      width="32px"
    />
  </button>
  <button
    className="main-navigation__arrow-button"
    type="button"
  >
    <img
      alt="search-icon"
      height="5px"
      src="/../../static/icons/arrow-down.svg"
      width="5px"
    />
  </button>
</header>
`;
